### PR TITLE
feat(data-table): custom cell implementation

### PR DESCRIPTION
**Pull request by Vignesh-Manogar-E3433**
#### Adding 2 predefined custom cell variants.
1. AnchorAnchor table
2. UserUser

#### Usage:
```js
let columns =  [{
      "key": "user",
      "text": "User",
      "orderIndex": 1,
      "variant": "user"
    }, {
      "key": "search",
      "text": "Preferred Search Engine",
      "orderIndex": 2,
      "variant": "anchor"
}];
let rows = [{
      "id": "001",
      "search": { "text": "Google", "href": "https://www.google.com" },
      "createdby": { 
        "image": "https://images.unsplash.com/photo-1614644147798-f8c0fc9da7f6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=300&q=80",
        "name": "Alexander Goodman", 
        "email": "alexander.goodman@freshdesk.com" 
      }
}];
```

#### Steps to run vuepress documentation:
- From root of project run npm run docs:dev
- Open the link that shows up in console after running above command.
- Navigate to DataTable component in sidebar.

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?
Tested by creating sample in Vuepress documentation.

### CODE REVIEW

File: packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  shadow: true,
+})
+export class CustomCellUser {
+  @Prop() image = null;
+
+  @Prop() name = '';
+
+  @Prop() email = '';
+
+  getInitialsFromName() {
```
Comment (parsuram-vijaysankar) [inline packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx:15]: I see this as a recurring pattern, Shall we move this to a util method, or better move this inside the avatar component itself?

File: packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Component, h, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'fw-custom-cell-anchor',
+  styleUrl: 'custom-cell-anchor.scss',
+  shadow: true,
+})
+export class CustomCellUser {
```
Comment (parsuram-vijaysankar) [inline packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx:8]: Why is the class named `CustomCellUser`?

File: packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.anchor {
```
Comment (parsuram-vijaysankar) [inline packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.scss:1]: Move this as a Global class/mixin inside typography? 
I think anchors might be the same across the application!

File: packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <div class='name-box-container'>
+        <div class='avatar'>
+          <fw-avatar
+            size='small'
+            initials={this.getInitialsFromName()}
+            image={this.image}
+          ></fw-avatar>
+        </div>
+        <div class='name-box'>
+          <div class='name-box-text'>{this.name}</div>
```
Comment (parsuram-vijaysankar) [inline packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx:36]: Would there be any truncation happening to content if the column size is smaller?
We need to review the details with the Design team.

File: packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx (hunk 1 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx]: Moved this logic to avatar component. Introduced name property to avatar component and displaying initials from name when image and initials props are not present.

File: packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.scss (hunk 3 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.scss:3]: Moved this to mixins and exposed a css class for anchor.

File: packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx (hunk 2 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx:8]: Named this as `CustomCellUser` to let users know that this displays user's information in a custom cell. kindly let me know if you feel we can improve on this with a more appropriate name.

File: packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx (hunk 4 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx:26]: As we discussed, currently column expands to the length of the content currently. But this logic will be changed. We will be adding 'min' and 'max-width' to columns. We will allow users to set width to a column by passing width as one of the properties of the column.

File: packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { Component, h, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'fw-custom-cell-user',
+  styleUrl: 'custom-cell-user.scss',
+  shadow: true,
+})
+export class CustomCellUser {
```
Comment (kaushal-freshworks) [inline packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx:8]: CustomCellUser class name is already present in the class above. Why have two classes with same name? Is that even allowed? Does this compile?

File: packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx (hunk 2 above)
Comment (kaushal-freshworks) [reply packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx:8]: Shouldnt this be "CustomCellAnchor" ?

Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/data-table/custom-cells/anchor/custom-cell-anchor.tsx]: Oh! my bad. Didn't notice the mistake before. I have changed this. Thanks.

File: packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx (hunk 5 above)
Comment (Vignesh-Manogar-E3433) [reply packages/crayons-core/src/components/data-table/custom-cells/user/custom-cell-user.tsx:8]: This was a mistake from my side. Changed 'CustomCellAnchor'. This compiled without issues, that is why I didn't notice the mistake last time.

**Review by parsuram-vijaysankar — APPROVED** [review]

**Pull request merged**